packages/contracts/contracts/core/permission/BulkPermissionsLib.sol

```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.10;
+
+/// @title BulkPermissionsLib
+/// @author Aragon Association - 2021, 2022
+/// @notice A library containing objects for bulk permission processing.
+library BulkPermissionsLib {
+    enum Operation {
+        Grant,
+        Revoke,
+        MakeImmutable
```

Comment:
This was discussed here already: https://github.com/aragon/core/pull/62#discussion_r923299041
@mathewmeconry Is also advocating for `makeImmutable`.

Reply:
> But "make immutable" suggests that you could leave the permission enabled forever.

This is precisely what you can do: Make permissions immutable so that you can never ever change them again.

`grant`, `_grantWithOracle`, and `revoke` all check `isImmutable` and revert if this it returns `true`. 

See here: 
https://github.com/aragon/core/blob/4b2da07414b022770252364293a4a385c3896e38/packages/contracts/contracts/core/permission/PermissionManager.sol#L234-L241
(note that `grant` calls `_grantWithOracle` internally.

https://github.com/aragon/core/blob/4b2da07414b022770252364293a4a385c3896e38/packages/contracts/contracts/core/permission/PermissionManager.sol#L261-L268